packages/network/src/logic/node/ProxyStreamConnectionManager.ts

```diff
@@ -0,0 +1,211 @@
+                logger.warn(reason)
+                return
+            }
+            this.addConnection(spid, targetNodeId)
+            await this.trackerManager.connectToSignallingOnlyTracker(trackerId, trackerAddress)
+            await promiseTimeout(this.nodeConnectTimeout, this.nodeToNode.connectToNode(targetNodeId, trackerId, false))
+            await this.nodeToNode.requestPublishOnlyStreamConnection(targetNodeId, spid)
+        } catch (err) {
+            logger.warn(`Failed to create a proxy outgoing stream connection to ${targetNodeId} for stream ${spid.key}:\n${err}`)
+            this.removeConnection(spid, targetNodeId)
+            this.node.emit(Event.PUBLISH_STREAM_REJECTED, targetNodeId, spid, err)
+        } finally {
+            this.trackerManager.disconnectFromSignallingOnlyTracker(trackerId)
+        }
+    }
+
+    async closeOutgoingStreamConnection(spid: SPID, targetNodeId: NodeId): Promise<void> {
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasOutOnlyConnection(spid, targetNodeId)) {
+            clearTimeout(this.getConnection(targetNodeId, spid)!.reconnectionTimer!)
+            await this.nodeToNode.leaveStreamOnNode(targetNodeId, spid)
+            this.removeConnection(spid, targetNodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, targetNodeId, spid)
+        } else {
+            logger.warn(`A proxy outgoing stream connection for ${spid.key} on node ${targetNodeId} does not exist`)
+        }
+    }
+
+    processLeaveRequest(message: UnsubscribeRequest, nodeId: NodeId): void {
+        const { streamId, streamPartition } = message
+        const spid = new SPID(streamId, streamPartition)
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasInOnlyConnection(spid, nodeId)) {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, nodeId, spid)
+        }
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasOutOnlyConnection(spid, nodeId)) {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, nodeId, spid)
+            logger.info(`Proxy node ${nodeId} closed one-way stream connection for ${spid}`)
+        }
+    }
+
+    async processPublishStreamRequest(message: PublishStreamConnectionRequest, nodeId: string): Promise<void> {
+        const { streamId, streamPartition } = message
+        const spid = new SPID(streamId, streamPartition)
+
+        // More conditions could be added here, ie. a list of acceptable ids or max limit for number of one-way this
+        const isAccepted = this.streamManager.isSetUp(spid) && this.acceptProxyConnections
+        if (isAccepted) {
+            this.streamManager.addInOnlyNeighbor(spid, nodeId)
+        }
+        return await this.nodeToNode.respondToPublishOnlyStreamConnectionRequest(nodeId, spid, isAccepted)
```

Comment:
generally `return` or `await`, but not both
```suggestion
        await this.nodeToNode.respondToPublishOnlyStreamConnectionRequest(nodeId, spid, isAccepted)
```